packages/truffle-compile/profiler/findUpdatedFiles.js

```diff
@@ -0,0 +1,45 @@
+const fse = require("fs-extra");
+
+const findUpdatedFiles = (
+  sourceFilesArtifacts,
+  sourceFilesArtifactsUpdatedTimes
+) => {
+  let updatedFiles = [];
+  // Stat all the source files, getting there updated times, and comparing them to
+  // the artifact updated times.
+  const sourceFiles = Object.keys(sourceFilesArtifacts);
+
+  let sourceFileStats;
+  sourceFileStats = sourceFiles.map(file => {
+    try {
+      return fse.statSync(file);
+    } catch (error) {
+      // Ignore it. This means the source file was removed
+      // but the artifact file possibly exists. Return null
+      // to signfy that we should ignore it.
+      return null;
+    }
+  });
+
+  sourceFiles.forEach((sourceFile, index) => {
```

Comment:
Can we turn this into a `.map` so we don't have to `updateFiles.push`?

Reply:
Oh yeah you bet we can!